tool-openssl/dgst.cc

```diff
@@ -0,0 +1,135 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <fcntl.h>
+#include <iostream>
+
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include "internal.h"
+
+// MD5 command currently only supports stdin
+static const argument_t kArguments[] = {
+    {"-help", kBooleanArgument, "Display option summary"},
+    {"-hmac", kOptionalArgument,
+     "Create a hashed MAC with the corresponding key"},
+    {"", kOptionalArgument, ""}};
+
+static bool dgst_file_op(const std::string &filename, const EVP_MD *digest,
+                         const std::string &hmac_key) {
+  ScopedFD scoped_fd = OpenFD(filename.c_str(), O_RDONLY | O_BINARY);
+  int fd = scoped_fd.get();
+
+  static const size_t kBufSize = 8192;
+  std::unique_ptr<uint8_t[]> buf(new uint8_t[kBufSize]);
+
+  if (!hmac_key.empty()) {
+    bssl::ScopedHMAC_CTX ctx;
+    if (!HMAC_Init_ex(ctx.get(), hmac_key.data(), hmac_key.size(), digest,
+                      nullptr)) {
+      fprintf(stderr, "Failed to initialize HMAC_Init_ex.\n");
+      return false;
+    }
+
+    // Update |buf| from file continuously.
+    for (;;) {
+      size_t n;
+      if (!ReadFromFD(fd, &n, buf.get(), kBufSize)) {
+        fprintf(stderr, "%s: No such file or directory\n", filename.c_str());
```

Comment:
NP: There was an error reading from the file, which doesn't necessarily mean that there's "no such file". (Could you even read from a directory?)

Reply:
Oh this was to align with OpenSSL's error messaging, I don't really enjoy it either.. I'm conflicted on compatibility everywhere or better error messages in general. 
I can change this to my original error message which also prints the `errno`. It's not really a 1 way door and we can change to align if we get complaints.